extensions/bouncer.md

```diff
@@ -0,0 +1,236 @@
+
+Replies:
+* Network changed:
+  - `[s] BOUNCER changenetwork netid RPL_OK`
+* Various errors:
+  - `[s] BOUNCER changenetwork * ERR_INVALIDARGS`
+  - `[s] BOUNCER changenetwork netid ERR_NETNOTFOUND`
+  - `[s] BOUNCER changenetwork netid ERR_INVALIDPORT`
+  - `[s] BOUNCER changenetwork netid ERR_UNKNOWN :Optional extra information`
+    
+
+
+### Removing a network
+Syntax: `[c] BOUNCER delnetwork <netid | *>`
+
+Delete a network from the bouncer. If deleting the active network the implimentation may choose how to handle the client connection. Examples include closing it or parting all channels and resorting it to a temporary connection.
+
+The bouncer MAY send the `LISTNETWORKS` reply to all the users connected clients to keep them in sync.
+
+Replies:
+* Network removed:
+  - `[s] BOUNCER delnetwork netid RPL_OK`
+* Various errors:
+  - `[s] BOUNCER delnetwork * ERR_INVALIDARGS`
+  - `[s] BOUNCER delnetwork netid ERR_NETNOTFOUND`
+  - `[s] BOUNCER delnetwork netid ERR_UNKNOWN :Optional extra information`
+
+
+## Network state notifications
+As a network state changes on the server such as connecting or disconnection, it MUST send the changes as they happen to any users connected clients.
+
+* Connection state changes:
+  - `[s] BOUNCER state netid netname connecting`
+  - `[s] BOUNCER state netid netname connected`
+  - `[s] BOUNCER state netid netname disconnected`
+
+## Network and buffer tags
+This document has described several tags that SHOULD be supported. Implementations MAY introduce other tags but MUST contain a vendor prefix, eg. `kiwibnc/mytag=value`.
+
+#### Network tags
+Implementations MUST recognise the following:
+- `network` - The human readable name for the network. It may be changed.
+- `state` - One of `connected`, `connecting`, or `disconnected` that represents the current connection to the IRC network.
+- `host` - The hostname to connect to the IRC network.
+- `port` - The port to connect to the IRC network.
+- `tls` - `1` if this IRC network requires TLS, `0` if not.
+- `tlsverify` - `1` if the bouncer should verify the TLS connection. `0` if not.
+- `password` - The server password.
+- `sasl_account` - The SASL account name to log into the IRC network with.
+- `sasl_pass` - The SASL password to log into the IRC network with.
+- `nick` - The nick for the active connection to the IRC network or the nick to use when connecting.
+- `username` - The username for the active connection to the IRC network or the username to use when connecting.
+- `realname` - The realname for the active connection to the IRC network or the realname to use when connecting.
+
+
+#### Buffer tags
+
+Implementations MUST recognise the following:
+- `network` - The name of the network this buffer belongs to.
+- `buffer` - The name of this buffer.
+- `seen` - The ISO 8601 timestamp of when the user last saw this buffer. Used to determine the point at which the user as read up to.
```

Comment:
This should be out-of-scope for the bouncer extension. Servers not implementing this spec (soju, pounce, oragono) may be interested in this functionality.